examples/dataclasses-avroschema-example/README.md

```diff
@@ -0,0 +1,105 @@
+## Example with dataclasses-avroschema
+
+[dataclasses-avroschema](https://github.com/marcosschroh/dataclasses-avroschema) example with `kstreams`
+
+### Requirements
+
+python 3.8+, poetry, docker-compose
+
+### Installation
+
+```bash
+poetry install
+```
+
+## Usage
+
+1. Start the kafka cluster: From `kstreams` project root execute `./scripts/cluster/start/`
+2. Inside the `dataclasses-avroschema-example` folder exeute `poetry run app`
+
+## Description
+
+In this example we want to stream instances of `models.User` and `models.Address` using `avro-schemas`:
+
+```python
+# models.py
+@dataclass
+class User(AvroModel):
+    "An User"
+    name: str
+    age: int
+    pets: typing.List[str]
+    accounts: typing.Dict[str, int]
+    country: str = "Argentina"
+    address: str = None
+
+    class Meta:
+        namespace = "User.v1"
+        aliases = ["user-v1", "super user",]
+
+
+@dataclass
+class Address(AvroModel):
+    "An Address"
+    street: str
+    street_number: int
+
+    class Meta:
+        namespace = "Address.v1"
+        aliases = ["address-v1",]
+```
+
+To accomplish it, we need to define custom `serializers` for our `models`:
+
+```python
+# serializers.py
+class AvroSerializer:
```

Comment:
Tecnically, you don't event need this, right?

```python
class KstreamSerializer:
     async def serialize(self, instance: AvroModel, **kwargs) -> bytes:
         ...
     async def deserialize(
        self, consumer_record: aiokafka.structs.ConsumerRecord, **kwargs
    ) -> aiokafka.structs.ConsumerRecord:
        ...

class KAvroModel(AvroModel, KstreamSerializer):
    ...

@dataclass
class User(KAvroModel):
     ...


@stream_engine.stream(
    user_topic, value_deserializer=User
)
async def user_stream(stream: Stream):
    async for cr in stream:
        print(f"Event consumed on topic {user_topic}. The user is {cr.value}")
```

Reply:
Your approach is incorrect:

1. `deserialize` is a `instance` method and you are using a class to make reference in the `stream`: value_deserializer=User. 
2. You have multiple inheritance and AvroModel defines `serialize` and `deserialize`, which means that `KstreamSerializer.serialize` and `KstreamSerializer.deserialize` are useless.